src/walker.rs

```diff
@@ -94,7 +103,21 @@ impl Walker {
       return Ok(Files::file(self.root, Bytes::from(root_metadata.len())));
     }
 
-    let filter = |entry: &walkdir::DirEntry| {
+    let mut file_infos = Vec::new();
+    let mut total_size = 0;
+
+    let mut walk_builder = WalkBuilder::new(&self.root);
+    walk_builder
+      .follow_links(self.follow_symlinks)
+      .standard_filters(false)
+      .hidden(!self.include_hidden);
+    if self.ignore_files {
+      walk_builder.ignore(true);
+      walk_builder.git_ignore(true);
```

Comment:
I think the flag should probably turn on pretty much all the ignore machinery, i.e.:

```
parents(true)
ignore(true)
git_ignore(true)
git_global(true)
git_exclude(true)
```